cpg_infra/billing_aggregator/aggregate/utils.py

```diff
@@ -797,15 +800,50 @@ def get_unit_for_batch_resource_type(batch_resource_type: str) -> str:
 
 
 def get_start_and_end_from_request(
-    request,
+    request: Request,
 ) -> tuple[datetime | None, datetime | None]:
     """
     Get the start and end times from the cloud function request.
     """
-    if request:
-        print(request)
-        # return request.params['start'], request.params['end']
-    return None, None
+    if not request:
+        return None, None
+
+    content_type = request.content_type
+    if request.method == 'GET':
+        print('GET request, using args')
+        request_data = request.args
+    elif content_type == 'application/json':
+        print('JSON found in request')
+        request_data = request.get_json(silent=True)
+    elif content_type in ('application/octet-stream', 'text/plain'):
+        print('Text data found')
+        request_data = json.loads(request.data)
+    elif content_type == 'application/x-www-form-urlencoded':
+        print('Encoded Form')
+        request_data = request.form
+    else:
+        raise ValueError(f'Unknown content type: {content_type}')
+
+    if 'attributes' in request_data and (
```

Comment:
Nitpick, but you don't need this `"attributes" in requests_data` as the below is defensive (`request_data.get('attributes')` won't crash)